app/Services/AdventuresLeagueAdapter.php

```diff
@@ -83,33 +86,77 @@ public function getCharacter($filePath): ?Character
     private function getEntries($data, $characterId)
     {
         $entries = [];
+
+        // entries only begin on line 5 (index 4) of the exported csv
         for ($i = 4; $i < count($data); $i++) {
-            //Check if row is a magic item entry
             $isItemEntry = ($data[$i][0] == "MAGIC ITEM");
+            $isCharacterLogEntry = ($data[$i][0] == "CharacterLogEntry");
+            $isCampaignLogEntry = ($data[$i][0] == "CampaignLogEntry");
+            $isDmLogEntry = ($data[$i][0] == "DmLogEntry");
+            $isTradeLogEntry = ($data[$i][0] == "TradeLogEntry");
+            $isPurchaseLogEntry = ($data[$i][0] == "PurchaseLogEntry");
+
+            $isEntryLine = ($isCharacterLogEntry || $isDmLogEntry || $isCampaignLogEntry || $isTradeLogEntry || $isPurchaseLogEntry);
 
-            if (!$isItemEntry) {
+            if ($isEntryLine) {
+                $type = "";
+                if ($isDmLogEntry) {
+                    $type = Entry::TYPE_DM;
+                } elseif ($isCampaignLogEntry || $isCharacterLogEntry) {
+                    $type = Entry::TYPE_GAME;
+                } else {
+                    $type = Entry::TYPE_DOWNTIME;
+                }
                 $entryData = [
                     'user_id' => Auth::id(),
+                    'adventure_id' => null,
+                    'campaign_id' => null,
                     'character_id' => $characterId,
-                    'date_played' => (array_key_exists(3, $data[$i])) ? $data[$i][3] : now(),
-                    'type' => Entry::TYPE_GAME,
+                    'event_id' => null,
+                    'dungeon_master_id' => null,
+                    'dungeon_master' => (array_key_exists(12, $data[$i])) ? (float)$data[$i][12] : "",
+                    'location' => (array_key_exists(11, $data[$i])) ? (float)$data[$i][11] : "",
+                    'levels' => 0,
+                    'date_played' => ($data[$i][3] == "") ? now() : $data[$i][3],
                     'gp' => (array_key_exists(7, $data[$i])) ? (float)$data[$i][7] : 0,
+                    'downtime' => (array_key_exists(8, $data[$i])) ? (float)$data[$i][8] : 0,
+                    'type' => $type,
                 ];
-                array_push($entries, new Entry($entryData));
-            }
-            // Item importing to be refactored
-            /*else {
-                //create the item and attach it to the last entry
-                $latestEntry = end($entries);
-                if (is_null($latestEntry)) {
-                    continue;
+                $entry = Entry::factory()->create($entryData);
+                $entry->save();
+                array_push($entries, $entry);
+            } else {
+                if (count($entries) == 0) {
+                    $e = Entry::factory()->create([
```

Comment:
Same as above, no need to call `factory()`
```suggestion
                    $e = Entry::create([
```